core/src/main/java/org/apache/calcite/config/CalciteSystemProperty.java

```diff
@@ -239,6 +239,12 @@
   public static final CalciteSystemProperty<Boolean> TEST_REDIS =
       booleanProperty("calcite.test.redis", true);
 
+  /**
+   * Whether to use Docker with TestContainers (https://www.testcontainers.org/) when available
+   */
+  public static final CalciteSystemProperty<Boolean> TEST_WITH_DOCKER_CONTAINER =
+      booleanProperty("calcite.test.docker", true);
+
```

Comment:
What's the benefit of having this property?
If Docker is available we can run the tests, if not then either we skip them or fallback to another execution method.

Reply:
@tgrall , would you please add a corresponding comment to the javadoc?

I believe the key point of having such a property is to allow running a non-docker test even in the case Docker is installed on the developer's machine.
For instance, it might be helpful to replicate an issue that happens only in non-docker test mode.